impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -243,6 +265,9 @@ public OnHeapStore(final Configuration<K, V> config, final TimeSource timeSource
       });
       if (result == null) {
         getObserver.end(StoreOperationOutcomes.GetOutcome.MISS);
+        if (byteSized && expiredValue.get() != null) {
+          decrementCurrentUsageInBytesIfRequired(getSizeOfKeyWithOffset(key) + getSizeOfValueWithOffset(expiredValue.get().value()) + sizeOfEngine.getCHMOffset());
```

Comment:
I don't think the size should be recalculated when the mapping gets evicted/removed/expired. This should be stored somewhere like in the ValueHolder.